(Post) Add team update for 02-28-2020 (#24)
puppetlabs/iac: _posts/2020-02-28-status-update.md

---
layout: post
title: "28th Feb 2020: IAC Team Status Update"
author: michaeltlombardi
categories:
  - team
  - status
tags:
  - module updates
  - community
  - litmus
---

The IAC team has been working on a lot of interesting things over the past week.
Here's a brief overview of what we've been up to:

## Updates to Puppet Modules

No modules were released since last week.

## Community Contributions

Thanks to [@jmangt](https://github.com/jmangt) for their improvements to the provision module,
[adding support for sssh'ing to vagrant boxes where root is not allowed to ssh](https://github.com/puppetlabs/provision/pull/103) and [ensuring the vagrant provision task works against paths with spaces in them](https://github.com/puppetlabs/provision/pull/101).

> <!-- ## Approved Modules

Author: I don't think any new modules were approved? If not, I can delete this section entirely.

[Puppet Approved](https://forge.puppet.com/approved) modules are recommended by us for use with Puppet Open Source and Puppet Enterprise and meet our expectations for quality and usability. -->

## Weekly Test Runs

Numerous PRs went through this week to ensure 44 modules (all those supported modules which have been converted to Litmus so far) run weekly tests via GitHub Actions.
These tests run against the master branch executing the full test suite - all of the spec tests and then all of the acceptance tests against the full matrix of operating systems for that module (defined in the `release_checks` section of the `provision.yaml` for that module).

As with the rest of our automated tests, these results are sent to Honeycomb for introspection later;
these test results are also recorded to the [Community Dashboard](https://puppetlabs.github.io/community_management/) in the `Github Actions Report` tab, which shows the test results for the last five runs for each workflow and module.
Right now, these results are recorded fairly simply, listing the run number, the branch the tests ran against, and whether they passed or failed.

> In the future, we will be improving the usefulness of this dashboard in numerous ways, but for now we just wanted a way to get some quick visualization for triage.

Author: I opted not to place any pictures here for now, but I could snap a particularly good image from the dashboard?

Reviewer: I'd leave honeycomb out for now here, as e.g. pass/fail status is currently not recorded. See https://tickets.puppetlabs.com/browse/IAC-537 for next steps.

## Facter 4 Incoming

We started doing some prep work this week as [Facter](https://github.com/puppetlabs/facter-ng)'s next major update has [become available as prerelease](https://rubygems.org/gems/facter/versions/4.0.8.pre).

> Facter 4 is a return to Ruby from C++ which should mean easier integration with the rest of our modules and tooling but also means we've got some checking and verifying to do as this major version rolls out.

Author: I wasn't sure what to highlight/cover here as I didn't work on any of this and didn't see much on it in scrollback.

> Facter 4 is a return to Ruby from C++ which should mean easier integration with the rest of our modules and tooling but also means we've got some checking and verifying to do as this major version rolls out.

Reviewer: ```suggestion
We started doing some prep work this week as [Facter](https://github.com/puppetlabs/facter-ng)'s next major update has [become available as prerelease](https://rubygems.org/gems/facter/versions/4.0.8.pre).
Facter 4 is a return to Ruby from C++ which allows deeper integration with the rest of our modules and test-tooling, and easier hacking on the core facts.
First we've got some checking and verifying to do as this major version rolls out to ensure a smooth experience for everyone.
If you want to try it out in your unit tests, specify `FACTER_GEM_VERSION='4.0.8.pre'` in your environment before any bundle or PDK commands.
Let us know in the [#puppet-dev](https://puppetcommunity.slack.com/archives/C0W1X7ZAL) channel on the [community slack](http://slack.puppet.com/) how it goes.
```

> We should have more on this in the near future, particularly any advice and guidance around what it will mean for you as module developers and consumers!

Reviewer: Release of the gem is planned for march 11th. cc @florindragos

> We should have more on this in the near future, particularly any advice and guidance around what it will mean for you as module developers and consumers!

Reviewer: ```suggestion
We should have more on this in the near future, particularly any advice and guidance around what it will mean for you as module developers and consumers!

## PuppetCamp Berlin 2020

In two weeks the sold out [PuppetCamp Berlin](https://www.eventbrite.com/e/puppet-camp-berlin-tickets-84908295999) will happen at the AWS OpsWorks offices, with a great showing of folks from the community and Puppet. If you're one of the lucky ones with a ticket, come and say hi to Melissa Casburn, our UX architect and [@DavidS](https://github.com/DavidS).
```

---
layout: post
title: "28th Feb 2020: IAC Team Status Update"
author: michaeltlombardi
categories:
  - team
  - status
tags:
  - module updates
  - community

>   - litmus

Reviewer: ```suggestion
  - litmus
  - facter
  - testing
  - pdk
```

This will carry the actual functionality for `pdk build` and allow third-party tools to re-use the same functionality without having to pull in the whole PDK.
A new Litmus release is ongoing to take advantage of this new development!

Thanks a lot to [Glenn Sarti](https://github.com/glennsarti) and [Florin Dragos](https://github.com/florindragos) for helping out with everything!

## Facter 4 Incoming

We started doing some prep work this week as [Facter](https://github.com/puppetlabs/facter-ng)'s next major update has [become available as prerelease](https://rubygems.org/gems/facter/versions/4.0.8.pre).
Facter 4 is a return to Ruby from C++ which allows deeper integration with the rest of our modules and test-tooling, and easier hacking on the core facts.
First we've got some checking and verifying to do as this major version rolls out to ensure a smooth experience for everyone.

> If you want to try it out in your unit tests, specify `FACTER_GEM_VERSION='4.0.8.pre'` in your environment before any bundle or PDK commands.

Reviewer: ```suggestion
If you want to try it out in your module's unit tests, specify `FACTER_GEM_VERSION='4.0.8.pre'` in your environment before any bundle or PDK commands.
```

Let us know in the [#puppet-dev](https://puppetcommunity.slack.com/archives/C0W1X7ZAL) channel on the [community slack](http://slack.puppet.com/) how it goes.

We should have more on this in the near future, particularly any advice and guidance around what it will mean for you as module developers and consumers - the expected release date is March 11th, 2020!

## PDK 1.17.0 Release

> A [new release of the PDK](https://github.com/puppetlabs/pdk/blob/master/CHANGELOG.md#v1170-2020-02-27) is here with initial support ofr control repos, system-level settings, and [numerous template updates](https://github.com/puppetlabs/pdk-templates/compare/1.16.0...1.17.0)!

Reviewer: ```suggestion
A [new release of the PDK](https://github.com/puppetlabs/pdk/blob/master/CHANGELOG.md#v1170-2020-02-27) is here with initial support for control repos, system-level settings, and [numerous template updates](https://github.com/puppetlabs/pdk-templates/compare/1.16.0...1.17.0)!
```
